### PR TITLE
Bump dart-sass to 1.46.0; Fix issues around types namespace

### PR DESCRIPTION
**Pull request by wondersloth**
- Bump dart-sass to latest
- Fixed things that broke.
- Fixed bug where `sass.types.Null.NULL` was not a valid value in `SassJsMap.set(key, value)`

*Open Questions*
- [x] Is `sass.types.Null.NULL` a valid key for `SassJsMap.set(key, value)`? Currently it still uses the same `assertSassValue` method from `index.js`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "node-sass-utils",
-  "version": "1.1.3",
+  "version": "1.1.4",
```
Comment (wondersloth) [inline package.json:3]: Maybe bump major since this is dropping node support.

**Review by eoneill — COMMENTED** [review]
Overall looks good, but I think a lot of the changes here can be simplified if we extract the type checking logic.

Something like...
```js
function getSassType(value) {
  if (value.typeOf) {
    return value.typeOf();
  }
  
  var constructor = value.constructor;

  return constructor.utilSassType = constructor.utilSassType || constructor.name
    .replace(/^_?(?:Sass|sass\.types\.)(\D+?)(?:ean)?\d*$/, '$1') // matches SassType, sass.types.Type, _SassType0; also strips the `ean` from `Boolean` (for consistency with `#typeOf()`)
    .toLowerCase();
}
```

then refactor `util.typeOf`/`util.typeOfConstructor` to be...
```js
typeOf: function(value) {
  return getSassType(value);
},
typeOfConstructor: function(constructor) {
  return getSassType({ constructor });
}
```

and finally, anywhere we were manually checking `value.constructor.name` (i.e. `assertSassValue`/`castToSass`) should now use `getSassType(value)`.

File: package.json [hunk 2]
```diff
@@ -17,15 +17,18 @@
     "url": "https://github.com/sass-eyeglass/node-sass-utils/issues"
   },
   "engines": {
-    "node": ">= 0.12"
+    "node": ">= 12.13"
   },
   "homepage": "https://github.com/sass-eyeglass/node-sass-utils",
+  "resolutions": {
+    "graceful-fs": "^4.2.9"
+  },
```
Comment (eoneill) [inline package.json:25]: Is this resolution still needed?

Comment (wondersloth) [reply package.json:25]: Yes we can remove it. I hadn't noticed that grunt has been removed.

**Review by eoneill — APPROVED** [review]

**Pull request merged**